Source/PipeSystem/PipeSystem/Comps/CompConvertToResource.cs

```diff
@@ -34,10 +34,29 @@ public Thing HeldThing
             get
             {
                 List<Thing> thingList = parent.Map.thingGrid.ThingsListAt(parent.Position);
-                for (int i = 0; i < thingList.Count; i++)
+
+                if (Props.category != null)
+                {
+                    List<ThingDef> categoryList = Props.category.childThingDefs;
+                    for (int j = 0; j < categoryList.Count; j++)
+                    {
+                        for (int i = 0; i < thingList.Count; i++)
+                        {
+                            if (thingList[i].def == categoryList[j])
+                            {
+                                Log.Message("Eating " + thingList[i].ToString() + " from category " + Props.category.ToString());
```

Comment:
Please comment out the logging (useful for dev, but can annoy players)

Reply:
That was on me, forgot to change that after depths of testing.

Commented out.